cmd/infra/aws/util/util.go

```diff
@@ -31,12 +32,26 @@ func NewConfig(credentialsFile, region string) *aws.Config {
 	return awsConfig
 }
 
-func NewRoute53Config(credentialsFile string) *aws.Config {
-	awsConfig := NewConfig(credentialsFile, "us-east-1")
-	awsConfig.Retryer = client.DefaultRetryer{
+// NewAWSRoute53Config generates an AWS config with slightly different Retryer timings
+func NewAWSRoute53Config(credentialsFile string, credKey string, credSecretKey string) *aws.Config {
+
+	awsRoute53Config := NewConfig(credentialsFile, credKey, credSecretKey, "us-east-1")
+	awsRoute53Config.Retryer = client.DefaultRetryer{
 		NumMaxRetries:    10,
 		MinRetryDelay:    5 * time.Second,
 		MinThrottleDelay: 10 * time.Second,
 	}
-	return awsConfig
+	return awsRoute53Config
+}
+
+// NewConfig allows support for a CredentialsFile or StaticCredential (UserKey & UserSecretKey)
+// [Default] if all values are provided is to use credentialsFile
+// This allows methods to be used by the CLI or vendored for other Go code
+func NewConfig(credentialsFile string, credKey string, credSecretKey string, region string) *aws.Config {
```

Comment:
fwiw I was thinking of a single function

```
func NewConfig(credentialsFile string, credKey string, credSecretKey string, region string) *aws.Config {
	var creds *credentials.Credentials
	creds = credentials.NewStaticCredentials(credKey, credSecretKey, "")
	if credentialsFile != "" {
		creds = credentials.NewSharedCredentials(credentialsFile, "default")
	}

	awsConfig := aws.NewConfig().WithRegion(region).WithCredentials(creds)
	awsConfig.Retryer = client.DefaultRetryer{
		NumMaxRetries:    10,
		MinRetryDelay:    5 * time.Second,
		MinThrottleDelay: 5 * time.Second,
	}

	return awsConfig
}
```

but yours impl works just fine as well

Reply:
I'll combing both this and the suggestion above.